clang/lib/Sema/SemaType.cpp

```diff
@@ -6490,6 +6490,15 @@ static void HandleBTFTypeTagAttribute(QualType &Type, const ParsedAttr &Attr,
                                       TypeProcessingState &State) {
   Sema &S = State.getSema();
 
+  // This attribute is only supported in C.
+  // FIXME: we should implement checkCommonAttributeFeatures() in SemaAttr.cpp
```

Comment:
Presumably this is pretty onerous to do?  Do we have an idea of what other attributes are going to fall into this same hole?

Reply:
Somewhat onerous given how few type attributes we have and how special type attributes tend to be.

The only other attribute in the same bucket is `lifetimebound` (which is a C++-only attribute rather than a C-only one). However, there are a pile of recent additions that are marked as `DeclOrTypeAttr` but have no handling whatsoever in SemaType.cpp (and so they probably aren't type attributes to begin with): `CountedByAttr`, `CountedByOrNullAttr`, `SizedByAttr`, and `SizedByOrNullAttr`.

I can handle `lifetimebound` in this PR or in a follow-up if you'd like. We should contact the authors of the other attributes and find out what the intent is for them rather than try to repair them as a drive-by.

Reply:
I would think `lifetimebound` here would be acceptable/easy enough?  The others, please just ping the authors and get them to fix it themselves.

Reply:
Oh! It turns out `lifetimebound` is not really possible to run into in C because the only way it's accepted in a type position is as a member function qualifier. It also has some further issues in C++: https://godbolt.org/z/czPa3x7nb

So I think we should file an issue for the confusion and not make changes in this PR for it. WDYT?

Reply:
https://github.com/llvm/llvm-project/issues/107249 for the bounds-safety related attributes
https://github.com/llvm/llvm-project/issues/107252 for the `lifetimebound` attribute